modules/cudabgsegm/src/mog2.cpp

```diff
@@ -105,10 +105,21 @@ class MOG2Impl CV_FINAL : public cuda::BackgroundSubtractorMOG2
     void setVarInit(double varInit) CV_OVERRIDE { constantsHost_.varInit_ = (float)varInit; }
 
     double getVarMin() const CV_OVERRIDE { return constantsHost_.varMin_; }
-    void setVarMin(double varMin) CV_OVERRIDE { constantsHost_.varMin_ = ::fminf((float)varMin, constantsHost_.varMax_); }
-
+    void setVarMin(double varMin) CV_OVERRIDE {
+        if (nframes_ == 0) {
+            constantsHost_.varMin_ = (float)varMin;
+        } else {
+            constantsHost_.varMin_ = ::fminf((float)varMin, constantsHost_.varMax_);
```

Comment:
I got it. The confusion of min and max should be managed by user outside, as the api has `getVarMin` and `getVarMax`。

Reply:
Got it. Since `constantsHost_` will be upload to gpu in `initialize`, i can just compare them in `initialize`. and keep the `setVarXXX` method simple.

in mog2.cu, the varMin_ and varMax_ will be compared again.
`
varnew = ::fmaxf(varnew, constants->varMin_);
varnew = ::fminf(varnew, constants->varMax_);
`